web3/providers/websocket.py

```diff
@@ -18,57 +16,42 @@
 RESTRICTED_WEBSOCKET_KWARGS = {'uri', 'loop'}
 
 
-def _start_event_loop(loop):
-    asyncio.set_event_loop(loop)
-    loop.run_forever()
-    loop.close()
-
-
-def _get_threaded_loop():
-    new_loop = asyncio.new_event_loop()
-    thread_loop = Thread(target=_start_event_loop, args=(new_loop,), daemon=True)
-    thread_loop.start()
-    return new_loop
-
-
 def get_default_endpoint():
     return os.environ.get('WEB3_WS_PROVIDER_URI', 'ws://127.0.0.1:8546')
 
 
-class PersistentWebSocket:
+class WebSocketConnection:
+    local = threading.local()
 
-    def __init__(self, endpoint_uri, loop, websocket_kwargs):
-        self.ws = None
+    def __init__(self, endpoint_uri, websocket_kwargs):
         self.endpoint_uri = endpoint_uri
-        self.loop = loop
         self.websocket_kwargs = websocket_kwargs
+        self.local.ws = None
```

Comment:
Nice improvement here!
Curious to know if this was giving an error without `threading.local`